File: client/src/app/shared/models/Booking.ts
import { Room } from './Room';
import { Freelancer } from './Freelancer';
import { Intern } from './Intern';

export interface Booking {
  id: number;
  date: string; // ISO date string
  startTime: string; // e.g., "09:00:00"
  endTime: string;
  room: Room;
  freelancer?: Freelancer | null;
  intern?: Intern | null;
  amountPaid: number;
  paid: boolean;
}

File: server/src/main/java/com/coworkingspace/server/models/Booking.java
package com.coworkingspace.server.models;

import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.time.LocalDate;
import java.time.LocalTime;

@Data
@NoArgsConstructor
@Entity
@Table(name = "bookings")
public class Booking {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    private LocalDate date;

    private LocalTime startTime;

    private LocalTime endTime;

    @ManyToOne
    @JoinColumn(name = "room_id")
    private Room room;

    @ManyToOne
    @JoinColumn(name = "freelancer_id")
    private Freelancer freelancer;

    @ManyToOne
    @JoinColumn(name = "intern_id")
    private Intern intern;

    private double amountPaid;

    private boolean paid;
}
